olive/workflows/run/run.py

```diff
@@ -153,6 +153,10 @@ def run_engine(config: RunConfig, data_root: str = None):
         config.engine.azureml_client_config = config.azureml_client
 
     engine = config.engine.create_engine()
+
+    # config file will be uploaded to AML job
+    engine.set_olive_config(config.to_json())
```

Comment:
would AzureMLSystem.olive_config = config.json() accomplish similar functionality? In this way, we need change the engine anymore.

Reply:
This looks good to me! I'll update with this.